controller/class.tapestry-controller.php

```diff
@@ -43,54 +43,86 @@ class TapestryController
         'INVALID_CHILD_NODE' => [
             'MESSAGE'   => 'Node is not a child of the tapestry',
             'STATUS'    => ['status' => 400]
+        ],
+        'INVALID_NEW_TAPESTRY' => [
+            'MESSAGE'   => 'Tapestry should have at max one root node when being created',
+            'STATUS'    => ['status' => 400]
+        ],
+        'INVALID_NEW_LINK' => [
+            'MESSAGE'   => 'New Tapestry link is invalid',
+            'STATUS'    => ['status' => 400]
+        ],
+        'POST_ID_ALREADY_SET' => [
+            'MESSAGE'   => 'PostID is set when creating a new Tapestry',
+            'STATUS'    => ['status' => 500]
         ]
     ];
     private $postId;
 
     /**
      * Constructor
      */
-    public function __construct($postId = 0)
+    public function __construct($postId)
```

Comment:
Why did you remove the default? This means this argument will be mandatory, so even new tapestries will need to pass in 0 to work.

Reply:
What I am trying to do is that if there is no `$postId` passed in or `$postId` is valid, I would set it to NULL. The methods will then check for the $postId.

Reply:
Set the parameter default to NULL.